mono_repo/lib/src/commands/init.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:args/command_runner.dart';
+import 'package:path/path.dart' as p;
+
+const _pubspecFileName = 'pubspec.yaml';
+const _pkgCfgFileName = 'mono_pkg.yaml';
+const _repoCfgFileName = 'mono_repo.yaml';
+const _recursiveScanFlag = 'recursive';
+
+const _commentText =
+    '# See with https://github.com/dart-lang/mono_repo for details on this file\n';
+
+class InitCommand extends Command<void> {
+  @override
+  String get name => 'init';
+
+  @override
+  String get description => 'Scaffold a new mono repo.';
+
+  @override
+  void run() => scaffold(p.current, globalResults[_recursiveScanFlag] as bool);
```

Comment:
This command should use its own arg parser and then use `argResults` here - see the `generate`command for an example. Then we can set the default as well (it should be `false`, imo).